### DISCUSSION TITLE
Update on the Release Post Sunbird ED 5.2.0.0 Release

### DISCUSSION THREAD

**Discussion by chitranshu-keshav** [Sunbird-Ed/Community#403] [General Discussion]
In the next release of Sunbird ED post 5.2.0.0, the plan is to upgrade the angular version from current 9.0 to 15.0. This is required to be done on priority as Sunbird ED(front-end Angular modules) currently is on an unsupported version. This change will also enable ease of contribution from new and existing contributors.

Since this will be a breaking change hence the next release will be a major release 6.0.0.0. Due to a major release there will not be any intermediate release post 5.2.0.0 hence the next Sunbird ED release will be 6.0.0.0. 

The timelines of 6.0.0.0 release will be finalized during the planning phase between 20th Feb and 3rd of March in consultation with all. Also the development/PRs for Release 6.0.0.0 needs to be on Angular version 15 otherwise merging them will not be possible. This is to keep you informed so that the upcoming work is planned accordingly. There will be some discussions scheduled next week to finalize the plan/timelines etc. on 6.0.0.0 release.

@vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @kiranharidas187
@sanyapawah @aayushkedia95 @ChakshuGautam @abhay-samagra 
@pkranga @mathewjpallan @mahesh-maney 
@pritamps @karn-nahata 
@coolbung @parthlawate
@kameshbhr @mmanimegalai @anandvarada
@SanthoshVasabhaktula @vrayulu 
@ashokreddy1208 @gandham-santhosh @rajeevsathish @vinukumar-vs @Krishnaj20 @swayangjit 
@pallakartheekreddy @rhwarrier @anandp504 @reshmi-nair @surendrasinghs @mmanimegalai @maheshkumargangula 
@madhucr @alok-os

**Comment by Prateek-slokam** [comment]
@chitranshu-keshav Sorry for a bit of delay in the response, I have a few questions to understand this,
1. Would the 6.0 version release be a quarter/month or even the duration of this needs a discussion and plan? Also asking this since there is now no interim release marked after 5.2.
2. Would this be a release of only angular updates or even the functionalities can be parallelly taken?
3. Would this have the BB merger too?
4. Would the other BB be also required to taken in some steps towards this or is only for ED?

@vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @kiranharidas187

**Comment by chitranshu-keshav** [comment]
@Prateek-slokam please find the answers to your questions below 

_Q1. Would the 6.0 version release be a quarter/month or even the duration of this needs a discussion and plan? Also asking this since there is now no interim release marked after 5.2.__ 
Due to the angular migration, intermediate release will not be possible hence we would move to 6.0.0.0 directly from 5.2.0.0. The time line is not yet final. This needs to be discussed and finalized by next week.

_Q2. Would this be a release of only angular updates or even the functionalities can be parallelly taken?_ 
There is no restriction on the functionalities that can go in this release. Only dependency is that all the development for this release should happen on the angular version 15

_Q3. Would this have the BB merger too?_
Plan is to take the latest versions of all the Building Blocks and do a full regression testing

_Q4.  Would the other BB be also required to taken in some steps towards this or is only for ED?_
There are dependencies on other building blocks which are being discussed with the building blocks to align the development and timeline of the same. The tickets are created and it could be referenced in ED issue tracker (Sprint "ED Rel 6.0.0.0 Sprint")

The current scope of the 6.0.0.0 release can be found [**here**](https://project-sunbird.atlassian.net/jira/software/c/projects/ED/boards/103/backlog?quickFilter=379). I would request to finalize the ML scope for 6.0.0.0 by 3rd of March so that community has the full visibility of ED scope for 6.0.0.0 . By updating the Sprint to "ED Rel 6.0.0.0 Sprint" it will show up on the release board

cc @vinukumar-vs @Krishnaj20 please add if you have any inputs

**Reply by Krishnaj20** [reply]
@Prateek-slokam - Can you let me know what do you mean by 'Would this have the BB merger too?'

**Reply by Prateek-slokam** [reply]
@Krishnaj20 I wanted to know if the BB pull would also be taken up as it is done in general stable releases. Chitranshu answered my query. I was just asking to understand that what all things are same as the general stable release.

**Comment by kiranharidas187** [comment]
@vinukumar-vs Any timelines for the Angular 15 upgrade for the app and portal?

**Comment by vinukumar-vs** [comment]
We are in the process of Angular upgrade of front-end application as part of release-6.0.0

> **For Portal**

Team has upgraded portal code to Angular 14. Please find the below branches of angular upgrade.
**Angular 12** - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/5.2.0_WorkingV12
**Angular 13** - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/6.0v13
**Angular 14** - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/6.0v14

We are going to production with Angular 14 by considering the release timeline & other dependent changes to adopt other modules. Testing is inprogress for Angular 13 & 14.
@rajeevsathish Share if any information is missing/incorrect 


> **For Mobile:**

Mobile app we are upgrading from Angular 9 to 14 along with Ionic upgrade.
**Angular 10** - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v10
**Angular 11** - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v11
**Angular 12** - TBU (InProgress)
**Angular 13** - TBU 
**Angular 14** - TBU 
Mobile upgrade is taking more time than what we anticipated. We are having cyclic issues of Babel while upgrading to Angular 11. The same issue again while upgrading to Angular 12(even after resolved in Angular 11).
@swayangjit share if any information is missing/incorrect 

While doing the process of Angular upgrade we have removed the unsupported modules which are handled by other BB's. Request other BB's also to support for smooth upgrade of Angular 14. We have already communicating through the discussion threads in the respective BB's for dependent modules.

@vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @kiranharidas187
@sanyapawah @aayushkedia95 @ChakshuGautam @abhay-samagra
@pkranga @mathewjpallan @mahesh-maney
@pritamps @karn-nahata
@coolbung @parthlawate
@SanthoshVasabhaktula @vrayulu
@ashokreddy1208 @gandham-santhosh @rajeevsathish @vinukumar-vs @Krishnaj20 @swayangjit
@pallakartheekreddy @rhwarrier @anandp504 @reshmi-nair @surendrasinghs @mmanimegalai @maheshkumargangula

**Comment by kiranharidas187** [comment]
@vinukumar-vs we have upgraded our npm packages to angular  15. Should we downgrade it to angular 14?

We can help in upgrading the ML modules once the basic modules are upgraded and pushed.

**Reply by chitranshu-keshav** [reply]
@vinukumar-vs can you please get back on the above

**Comment by swayangjit** [comment]
@kiranharidas187 Angular 12 migration branch is ready for contribution. You can use the branch [release-6.0.0-v12](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v12)

Angular 13 migration will be ready for contribution by the end of this week

cc:- @Krishnaj20 @vinukumar-vs @chitranshu-keshav

**Comment by swayangjit** [comment]
@kiranharidas187 Angular 13 migration branch is ready for contribution. You can use the branch [release-6.0.0-v13](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v13)


cc:- @Krishnaj20 @vinukumar-vs @chitranshu-keshav

**Comment by vinukumar-vs** [comment]
> Mobile app (Sunbird-ED)

We are upgrading from Angular 9 to 13 along with Ionic upgrade. (we are going with only Angular-13 in release-6.0 for mobile app because of [Ionic 7](https://ionicframework.com/docs/updating/7-0) & Capacitor changes. Also Angular 14 is not fully supported/tested by Ionic).

Angular 10 - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v10
Angular 11 - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v11
Angular 12 - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v12
Angular 13 - https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/tree/release-6.0.0-v13

> Portal (Sunbird-ED)

We already updated the Angular branches of portal(Going with Angular 14 in release-6.0). Sharing the Github branch details again here
Angular 12 - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/5.2.0_WorkingV12
Angular 13 - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/6.0v13
Angular 14 - https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/6.0v14

@vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @kiranharidas187
@sanyapawah @aayushkedia95 @ChakshuGautam @abhay-samagra
@pkranga @mathewjpallan @mahesh-maney
@pritamps @karn-nahata
@coolbung @parthlawate
@SanthoshVasabhaktula @vrayulu
@ashokreddy1208 @gandham-santhosh @rajeevsathish @vinukumar-vs @Krishnaj20 @swayangjit
@pallakartheekreddy @rhwarrier @anandp504 @reshmi-nair @surendrasinghs @mmanimegalai @maheshkumargangula
